Project/Gem/Source/Navigation/SplinePosesPublisher.cpp

```diff
@@ -0,0 +1,141 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "SplinePosesPublisher.h"
+#include "AzCore/Math/MathUtils.h"
+#include "AzCore/Math/Vector3.h"
+#include "AzCore/std/containers/vector.h"
+#include "AzCore/std/string/string.h"
+#include <AzCore/Asset/AssetSerializer.h>
+#include <AzCore/Math/Matrix3x3.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <LmbrCentral/Shape/SplineComponentBus.h>
+#include <rclcpp/qos.hpp>
+
+namespace ROS2::Demo
+{
+
+    void SplinePosesPublisher::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<SplinePosesPublisher>()
+                ->Version(1)
+
+                ->Field("ReverseDirection", &SplinePosesPublisher::m_reverseDirection)
+                ->Field("PathTopicConfig", &SplinePosesPublisher::m_topicName)
+                ->Field("PoseCount", &SplinePosesPublisher::m_poseCount);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<SplinePosesPublisher>("SplinePosesPublisher", "SplinePosesPublisher")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &SplinePosesPublisher::m_reverseDirection,
+                        "Reverse Direction",
+                        "Reverse the robot movement direction")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &SplinePosesPublisher::m_topicName, "Topic Name", "Topic name to publish poses")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &SplinePosesPublisher::m_poseCount,
+                        "Pose Count",
+                        "Number of poses to create and publish");
+            }
+        }
+    }
+
+    void SplinePosesPublisher::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("SplineService"));
+    }
+
+    void SplinePosesPublisher::Activate()
+    {
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+
+        CalculatePoses();
+
+        rclcpp::QoS qos(5);
+        qos.reliability(rclcpp::ReliabilityPolicy::Reliable).durability(rclcpp::DurabilityPolicy::TransientLocal);
+
+        m_pathPublisher = ros2Node->create_publisher<nav_msgs::msg::Path>(m_topicName.data(), qos);
+        m_pathPublisher->publish(m_path);
+    }
+
+    void SplinePosesPublisher::Deactivate()
+    {
+    }
+
+    void SplinePosesPublisher::CalculatePoses()
```

Comment:
I think that this function should return the value of m_path instead of saving it in a private value.